server/routes/bitbucket/bitbucket_webhook.py

```
@@ -0,0 +1,280 @@
+    # supplied: allowlist-sanitize before they can reach any log line.
+    delivery_id = _safe_log_value(request.headers.get(DELIVERY_HEADER, "").strip())
+    event_type = _safe_log_value(request.headers.get(EVENT_HEADER, "").strip(), 40)
+    if not delivery_id or not event_type:
+        return _reject("missing_metadata", "missing webhook metadata", 400)
+    # Bitbucket retries reuse the same X-Request-UUID; namespace by org so
+    # a delivery id can never collide across providers/orgs in the shared
+    # webhook_deliveries table (column is VARCHAR(64); uuid+prefix fits).
+    return ((f"bb:{delivery_id}"[:64], event_type), None)
+
+
+def _record_and_dispatch(
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_record_and_dispatch`

(ANN202)